src/pages/AddCertificate/AddCertificate.tsx

```diff
@@ -0,0 +1,248 @@
+  validTo: string;
+  validRange: string;
+};
+
+type NewFormError = {
+  supplier: string;
+  certificateType: string;
+  validFrom: string;
+  validTo: string;
+  validRange: string;
+};
+
+const AddCertificate: React.FC = () => {
+  const navigate = useNavigate();
+  const [certificateData, setCertificateData] = useState<Certificate>({
+    supplier: '',
+    certificateType: '',
+    validFrom: null,
+    validTo: null,
+  });
+
+  const [pdfPreview, setPdfPreview] = useState<string | null>(null);
+  const [errors, setErrors] = useState<FormError>({
+    supplier: '',
+    certificateType: '',
+    validFrom: '',
+    validTo: '',
+    validRange: '',
+  });
+
+  const validateForm = (): boolean => {
+    const newErrors: NewFormError = {
+      supplier: '',
+      certificateType: '',
+      validFrom: '',
+      validTo: '',
+      validRange: '',
+    };
+
+    console.log('Validating form data:', certificateData);
+
+    if (!certificateData.supplier) newErrors.supplier = 'Supplier is required.';
+    if (!certificateData.certificateType)
+      newErrors.certificateType = 'Certificate type is required.';
+    if (!certificateData.validFrom)
+      newErrors.validFrom = 'Valid from date is required.';
+    if (!certificateData.validTo)
+      newErrors.validTo = 'Valid to date is required.';
+    if (
+      certificateData.validFrom &&
+      certificateData.validTo &&
+      certificateData.validFrom > certificateData.validTo
+    ) {
+      newErrors.validRange =
+        'Valid from date cannot be after the valid to date.';
+    }
+
+    console.log('Validation errors:', newErrors);
+
+    setErrors(newErrors);
+
+    const isValid = Object.values(newErrors).every((error) => error === '');
+    console.log('Form is valid:', isValid);
+
+    return isValid;
+  };
+
+  const handleInput = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+  ) => {
+    const { name, value } = e.target;
+    setCertificateData((prevData) => ({
+      ...prevData,
+      [name]: value,
+    }));
+  };
+
+  const handleDateChange = (name: string, date: Date | null) => {
+    setCertificateData((prevData) => ({
+      ...prevData,
+      [name]: date,
+    }));
+  };
+
+  const handleSave = () => {
+    console.log('Attempting to save...');
+    if (validateForm()) {
+      console.log('Form validated successfully');
+      setCertificateData(certificateData);
+      certificates.push(certificateData);
+      console.log('Certificate added:', certificateData);
+      console.log('Updated certificates array:', certificates);
+
+      setErrors({
+        supplier: '',
+        certificateType: '',
+        validFrom: '',
+        validTo: '',
+        validRange: '',
+      });
+
+      navigate('/machine-learning/example1');
+    } else {
+      console.log('Form validation failed');
+      console.log('Current errors:', errors);
+    }
+  };
+
+  const handleReset = () => {
+    setCertificateData({
+      supplier: '',
```

Comment:
please create a constant outside of component for intitial data, and then use it on all places where you need this initial state of certificate

Reply:
not fully resolved yet, please create constant outside of component, and then use it on all places where you need initialState